### PR TITLE
test codegen for cdcmp

### PR DESCRIPTION
**Pull request by MartinNowak**
will need a rebase after https://github.com/dlang/dmd/pull/7848

### CODE REVIEW

File: test/runnable/test_cdcmp.d [hunk 1]
```diff
@@ -0,0 +1,1477 @@
...
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
+        /* xor    eax,eax                 */ 0x31, 0xc0,
+        /* pop    rbp                     */ 0x5d,
+        /* ret                            */ 0xc3,
+    ]),
+    Code!(ubyte, "<", ubyte)([
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
+        /* sub    rsp,0x10                */ 0x48, 0x83, 0xec, 0x10,
```
Comment (WalterBright) [inline test/runnable/test_cdcmp.d]: All the 0xNN data can get incredibly tedious to create. In the future, use hexStrings -
```
0x48, 0x83, 0xEC, 0x10,
```
becomes:
```
hexString!"48 83 EC 10"
```

Comment (wilzbach) [reply test/runnable/test_cdcmp.d]: This file is auto-generated ;-)

File: test/runnable/test_cdcmp.d [hunk 2]
```diff
@@ -0,0 +1,1477 @@
...
+    import std.file : readText;
+    import std.format : formattedWrite;
+    import std.meta : AliasSeq;
+    import std.path : baseName, setExtension;
+    import std.process : environment, execute, pipeProcess, wait;
+    import std.range : dropOne;
+    import std.regex : ctRegex, matchFirst, replaceFirstInto;
+    import std.stdio : File, stdout, writeln;
+    import std.string : strip;
+    import std.typecons : tuple, EnumMembers;
```
[the remark this answers is not in the document]
Comment (wilzbach) [reply test/runnable/test_cdcmp.d]: No. It's just to automatically update the code blocks. Note `version(update)` block.
We have a few of these files already in the testsuite ;-)

**Review by wilzbach — APPROVED** [review]
Ready whenever you are.

File: test/runnable/test_cdcmp.d [hunk 3]
```diff
@@ -0,0 +1,1477 @@
...
+alias AliasSeq(Args...) = Args;
+
+// dfmt off
+alias baselineCases = AliasSeq!(
+    Code!(ubyte, "<", Zero!ubyte)([
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
+        /* xor    eax,eax                 */ 0x31, 0xc0,
+        /* pop    rbp                     */ 0x5d,
+        /* ret                            */ 0xc3,
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply test/runnable/test_cdcmp.d]: No, I didn't notice it was auto-generated. Thanks for letting me know.

File: test/runnable/test_cdcmp.d (hunk 2 above)
[the remark this answers is not in the document]
Comment (WalterBright) [reply test/runnable/test_cdcmp.d]: ok

**Comment by MartinNowak** [comment]
We should look to create some small tools/scripts to reduce boilerplate for those tests, maybe incorporating into d_do_test or make it a sidekick.
It turns out useful to review codegen changes (as the recent `int < 0` one), and improves confidence in doing those changes (here is the diff for the `int < 0` fix aab2201989f84fe1ad9b0e0bfff2f8df6fea879b).

File: test/runnable/test_cdcmp.d (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinNowak) [reply test/runnable/test_cdcmp.d]: > A better approach is to do it like test/runnable/iasm.d does, where a pointer to the start of the function is taken and then the generated bytes are compared.

That's how it's done for the testees, it was just to tedious to update the expectated outcome, so I added an automated update part using `objdump`. This allows easy diffing and turned out very helpful for the cdvecfill changes where you can see a commit-by-commit diff (https://github.com/dlang/dmd/pull/7003/commits).

File: test/runnable/test_cdcmp.d (hunk 1 above)
Comment (MartinNowak) [reply test/runnable/test_cdcmp.d]: HexString relies on phobos, so I avoided that.

**Pull request merged**